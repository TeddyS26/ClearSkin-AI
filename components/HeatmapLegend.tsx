import { View, Text } from "react-native";

export default function HeatmapLegend({ mode }: { mode: "breakouts"|"oiliness"|"dryness"|"redness" }) {
  const label = (() => {
    switch (mode) {
      case "breakouts": return "Green = clear, Red = more breakouts";
      case "oiliness":  return "Green = less oily, Red = very oily";
      case "dryness":   return "Green = hydrated, Red = very dry";
      case "redness":   return "Green = low redness, Red = high redness";
    }
  })();

  return (
    <View className="mt-2">
      <View className="flex-row items-center justify-between mb-2">
        <Text className="text-xs text-gray-600 font-medium">Good</Text>
        <View className="flex-row flex-1 mx-2 h-2 rounded-full overflow-hidden">
          <View className="flex-1" style={{ backgroundColor: "rgba(0,200,0,0.5)" }} />
          <View className="flex-1" style={{ backgroundColor: "rgba(255,215,0,0.5)" }} />
          <View className="flex-1" style={{ backgroundColor: "rgba(255,140,0,0.5)" }} />
          <View className="flex-1" style={{ backgroundColor: "rgba(255,0,0,0.5)" }} />
        </View>
        <Text className="text-xs text-gray-600 font-medium">Severe</Text>
      </View>
      <Text className="text-xs text-gray-500 text-center">{label}</Text>
    </View>
  );
}
