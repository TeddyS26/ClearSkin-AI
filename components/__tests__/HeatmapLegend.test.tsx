import React from "react";
import { render } from "@testing-library/react-native";
import HeatmapLegend from "../HeatmapLegend";

describe("HeatmapLegend", () => {
  it("should render with breakouts mode", () => {
    const { getByText } = render(<HeatmapLegend mode="breakouts" />);

    expect(getByText("Good")).toBeTruthy();
    expect(getByText("Severe")).toBeTruthy();
    expect(getByText("Green = clear, Red = more breakouts")).toBeTruthy();
  });

  it("should render with oiliness mode", () => {
    const { getByText } = render(<HeatmapLegend mode="oiliness" />);

    expect(getByText("Good")).toBeTruthy();
    expect(getByText("Severe")).toBeTruthy();
    expect(getByText("Green = less oily, Red = very oily")).toBeTruthy();
  });

  it("should render with dryness mode", () => {
    const { getByText } = render(<HeatmapLegend mode="dryness" />);

    expect(getByText("Good")).toBeTruthy();
    expect(getByText("Severe")).toBeTruthy();
    expect(getByText("Green = hydrated, Red = very dry")).toBeTruthy();
  });

  it("should render with redness mode", () => {
    const { getByText } = render(<HeatmapLegend mode="redness" />);

    expect(getByText("Good")).toBeTruthy();
    expect(getByText("Severe")).toBeTruthy();
    expect(getByText("Green = low redness, Red = high redness")).toBeTruthy();
  });

  it("should render gradient color bars", () => {
    const { toJSON } = render(<HeatmapLegend mode="breakouts" />);
    const tree = toJSON();

    expect(tree).toBeTruthy();
  });

  it("should display correct label for breakouts", () => {
    const { queryByText } = render(<HeatmapLegend mode="breakouts" />);

    expect(queryByText("Green = clear, Red = more breakouts")).toBeTruthy();
    expect(queryByText("Green = less oily, Red = very oily")).toBeFalsy();
    expect(queryByText("Green = hydrated, Red = very dry")).toBeFalsy();
    expect(queryByText("Green = low redness, Red = high redness")).toBeFalsy();
  });

  it("should display correct label for oiliness", () => {
    const { queryByText } = render(<HeatmapLegend mode="oiliness" />);

    expect(queryByText("Green = less oily, Red = very oily")).toBeTruthy();
    expect(queryByText("Green = clear, Red = more breakouts")).toBeFalsy();
    expect(queryByText("Green = hydrated, Red = very dry")).toBeFalsy();
    expect(queryByText("Green = low redness, Red = high redness")).toBeFalsy();
  });

  it("should display correct label for dryness", () => {
    const { queryByText } = render(<HeatmapLegend mode="dryness" />);

    expect(queryByText("Green = hydrated, Red = very dry")).toBeTruthy();
    expect(queryByText("Green = clear, Red = more breakouts")).toBeFalsy();
    expect(queryByText("Green = less oily, Red = very oily")).toBeFalsy();
    expect(queryByText("Green = low redness, Red = high redness")).toBeFalsy();
  });

  it("should display correct label for redness", () => {
    const { queryByText } = render(<HeatmapLegend mode="redness" />);

    expect(queryByText("Green = low redness, Red = high redness")).toBeTruthy();
    expect(queryByText("Green = clear, Red = more breakouts")).toBeFalsy();
    expect(queryByText("Green = less oily, Red = very oily")).toBeFalsy();
    expect(queryByText("Green = hydrated, Red = very dry")).toBeFalsy();
  });

  it("should match snapshot for breakouts mode", () => {
    const { toJSON } = render(<HeatmapLegend mode="breakouts" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("should match snapshot for oiliness mode", () => {
    const { toJSON } = render(<HeatmapLegend mode="oiliness" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("should match snapshot for dryness mode", () => {
    const { toJSON } = render(<HeatmapLegend mode="dryness" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("should match snapshot for redness mode", () => {
    const { toJSON } = render(<HeatmapLegend mode="redness" />);
    expect(toJSON()).toMatchSnapshot();
  });
});

