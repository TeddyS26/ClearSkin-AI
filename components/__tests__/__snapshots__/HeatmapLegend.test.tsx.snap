// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeatmapLegend should match snapshot for breakouts mode 1`] = `
<View
  className="mt-2"
>
  <View
    className="flex-row items-center justify-between mb-2"
  >
    <Text
      className="text-xs text-gray-600 font-medium"
    >
      Good
    </Text>
    <View
      className="flex-row flex-1 mx-2 h-2 rounded-full overflow-hidden"
    >
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(0,200,0,0.5)",
          }
        }
      />
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(255,215,0,0.5)",
          }
        }
      />
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(255,140,0,0.5)",
          }
        }
      />
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(255,0,0,0.5)",
          }
        }
      />
    </View>
    <Text
      className="text-xs text-gray-600 font-medium"
    >
      Severe
    </Text>
  </View>
  <Text
    className="text-xs text-gray-500 text-center"
  >
    Green = clear, Red = more breakouts
  </Text>
</View>
`;

exports[`HeatmapLegend should match snapshot for dryness mode 1`] = `
<View
  className="mt-2"
>
  <View
    className="flex-row items-center justify-between mb-2"
  >
    <Text
      className="text-xs text-gray-600 font-medium"
    >
      Good
    </Text>
    <View
      className="flex-row flex-1 mx-2 h-2 rounded-full overflow-hidden"
    >
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(0,200,0,0.5)",
          }
        }
      />
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(255,215,0,0.5)",
          }
        }
      />
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(255,140,0,0.5)",
          }
        }
      />
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(255,0,0,0.5)",
          }
        }
      />
    </View>
    <Text
      className="text-xs text-gray-600 font-medium"
    >
      Severe
    </Text>
  </View>
  <Text
    className="text-xs text-gray-500 text-center"
  >
    Green = hydrated, Red = very dry
  </Text>
</View>
`;

exports[`HeatmapLegend should match snapshot for oiliness mode 1`] = `
<View
  className="mt-2"
>
  <View
    className="flex-row items-center justify-between mb-2"
  >
    <Text
      className="text-xs text-gray-600 font-medium"
    >
      Good
    </Text>
    <View
      className="flex-row flex-1 mx-2 h-2 rounded-full overflow-hidden"
    >
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(0,200,0,0.5)",
          }
        }
      />
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(255,215,0,0.5)",
          }
        }
      />
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(255,140,0,0.5)",
          }
        }
      />
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(255,0,0,0.5)",
          }
        }
      />
    </View>
    <Text
      className="text-xs text-gray-600 font-medium"
    >
      Severe
    </Text>
  </View>
  <Text
    className="text-xs text-gray-500 text-center"
  >
    Green = less oily, Red = very oily
  </Text>
</View>
`;

exports[`HeatmapLegend should match snapshot for redness mode 1`] = `
<View
  className="mt-2"
>
  <View
    className="flex-row items-center justify-between mb-2"
  >
    <Text
      className="text-xs text-gray-600 font-medium"
    >
      Good
    </Text>
    <View
      className="flex-row flex-1 mx-2 h-2 rounded-full overflow-hidden"
    >
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(0,200,0,0.5)",
          }
        }
      />
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(255,215,0,0.5)",
          }
        }
      />
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(255,140,0,0.5)",
          }
        }
      />
      <View
        className="flex-1"
        style={
          {
            "backgroundColor": "rgba(255,0,0,0.5)",
          }
        }
      />
    </View>
    <Text
      className="text-xs text-gray-600 font-medium"
    >
      Severe
    </Text>
  </View>
  <Text
    className="text-xs text-gray-500 text-center"
  >
    Green = low redness, Red = high redness
  </Text>
</View>
`;
